src/compilation/compiler.rs

```diff
@@ -215,24 +217,28 @@ impl Compiler {
 
     /// Registers a function prototype, panicking if the function has already been defined.
     unsafe fn compile_func_proto(&mut self, func_proto: &FuncProto) {
-        if self.scope_manager.get_func(&func_proto.name).is_some() {
-            panic!("Cannot redefine function '{}'", func_proto.name);
+        if self.scope_manager.get(&func_proto.name).is_some() {
+            panic!("Cannot redefine value '{}'", func_proto.name);
```

Comment:
I feel like the error message `cannot redefine 'print'` is a little nicer than `cannot redefine value 'print'`. Thoughts?

```suggestion
        if self.scope_manager.get(&func_proto.name).is_some() {
            panic!("Cannot redefine '{}'", func_proto.name);
```